contracts/optimistic-ethereum/OVM/bridge/tokens/Abs_L2DepositedToken.sol

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+pragma experimental ABIEncoderV2;
+
+/* Interface Imports */
+import { iOVM_L2DepositedToken } from "../../../iOVM/bridge/tokens/iOVM_L2DepositedToken.sol";
+import { iOVM_L1TokenGateway } from "../../../iOVM/bridge/tokens/iOVM_L1TokenGateway.sol";
+
+/* Library Imports */
+import { OVM_CrossDomainEnabled } from "../../../libraries/bridge/OVM_CrossDomainEnabled.sol";
+
+/**
+ * @title Abs_L2DepositedToken
+ * @dev An L2 Deposited Token is an L2 representation of funds which were deposited from L1.
+ * Usually contract mints new tokens when it hears about deposits into the L1 ERC20 gateway.
+ * This contract also burns the tokens intended for withdrawal, informing the L1 gateway to release L1 funds.
+ *
+ * NOTE: This abstract contract gives all the core functionality of a deposited token implementation except for the
+ * token's internal accounting itself.  This gives developers an easy way to implement children with their own token code.
+ *
+ * Compiler used: optimistic-solc
+ * Runtime target: OVM
+ */
+abstract contract Abs_L2DepositedToken is iOVM_L2DepositedToken, OVM_CrossDomainEnabled {
+
+    /*******************
+     * Contract Events *
+     *******************/
+
+    event Initialized(iOVM_L1TokenGateway _l1TokenGateway);
+
+    /********************************
+     * External Contract References *
+     ********************************/
+
+    iOVM_L1TokenGateway public l1TokenGateway;
+
+    /********************************
+     * Constructor & Initialization *
+     ********************************/
+
+    /**
+     * @param _l2CrossDomainMessenger L1 Messenger address being used for cross-chain communications.
+     */
+    constructor(
+        address _l2CrossDomainMessenger
+    )
+        OVM_CrossDomainEnabled(_l2CrossDomainMessenger)
+    {}
+
+    /**
+     * @dev Initialize this contract with the L1 token gateway address.
+     * The flow: 1) this contract gets deployed on L2, 2) the L1
+     * gateway is dpeloyed with addr from (1), 3) L1 gatweway address passed here.
+     *
+     * @param _l1TokenGateway Address of the corresponding L1 gateway deployed to the main chain
+     */
+
+    function init(
+        iOVM_L1TokenGateway _l1TokenGateway
+    )
+        public
+    {
+        require(address(l1TokenGateway) == address(0), "Contract has already been initialized");
+
+        l1TokenGateway = _l1TokenGateway;
+        
+        emit Initialized(l1TokenGateway);
```

Comment:
wait does this need to be either `emit Initialized(_l1TokenGateway);` or `emit Initialized(address(l1TokenGateway));`